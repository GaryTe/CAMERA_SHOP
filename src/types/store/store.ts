import { store } from '../../store/store/store';
import { listCamera, CardProduct, Reviews, PromoProduct, Review } from '../types-response/types-response';

export type Cameras = {
  cameras: listCamera | null;
  modifiedListProduct: listCamera | null;
}

export type CardCamera = {
  dataCamera: CardProduct | null;
}

export type SimilaCameras = {
  dataSimilaCameras: listCamera | null;
  sortListCamerra: listCamera | null;
  counterMin: number;
  counterMax: number;
}

export type ReviewsCamera = {
  dataReviews: Reviews | Array<Review>;
  resultSortReviews: Reviews | null;
  reviews: Reviews | null;
  counter: number;
}

export type PromoCamera = {
  dataPromoCamera: PromoProduct | null;
}

export type Cart = {
  dataProduct: listCamera | Array<CardProduct>;
  loading: boolean;
}

export type RootState = ReturnType<typeof store.getState>;
