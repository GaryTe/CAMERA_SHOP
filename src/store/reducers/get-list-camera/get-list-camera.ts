import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { fetchDataListCamera } from '../../api-actions/api-actions';
import { down } from '../../../util/util';
import { Cameras } from '../../../types/store/store';
import { NameSpace, stringValue } from '../../../const/const';
import { listCamera } from '../../../types/types-response/types-response';
import { listProduct } from '../../../util/mocks';

const initialState: Cameras = {
  cameras: null,
  modifiedListProduct: null,
};

export const getListCamera = createSlice({
  name: NameSpace.ListCamera,
  initialState,
  reducers: {
    changeDataProduct: (state, action: PayloadAction<listCamera>) => {
      state.modifiedListProduct = action.payload;
    },
    functionForTesting: (state) => {
      state.cameras = listProduct();
      state.modifiedListProduct = listProduct();
    }
  },

  extraReducers(builder) {
    builder
      .addCase(fetchDataListCamera.fulfilled, (state, action) => {
        state.cameras = action.payload;
        state.modifiedListProduct = down(action.payload, stringValue[0]);
      });
  }
});

export const {changeDataProduct, functionForTesting} = getListCamera.actions;
