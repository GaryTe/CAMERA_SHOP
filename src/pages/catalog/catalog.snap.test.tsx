import {render} from '@testing-library/react';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import Catalog from './catalog';
import { store } from '../../store/store/store';
import { functionForTesting } from '../../store/reducers/get-list-camera/get-list-camera';

const history = createMemoryHistory();

const children = <Catalog />;

describe('Component: Catalog', () => {
  it('should render Catalog', () => {
    store.dispatch(functionForTesting());

    const {container} = render(
      <Provider store={store}>
        <Router
          location={history.location}
          navigationType={history.action}
          navigator={history}
        >
          {children}
        </Router>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
