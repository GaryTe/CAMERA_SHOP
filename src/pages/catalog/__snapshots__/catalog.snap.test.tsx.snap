// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Catalog should render Catalog 1`] = `
<div>
  <header
    class="header"
    id="header"
  >
    <div
      class="container"
    >
      <a
        aria-label="Переход на главную"
        class="header__logo"
        href="/"
      >
        <svg
          aria-hidden="true"
          height="36"
          width="100"
        >
          <use
            xlink:href="#icon-logo"
          />
        </svg>
      </a>
      <nav
        class="main-nav header__main-nav"
      >
        <ul
          class="main-nav__list"
        >
          <li
            class="main-nav__item"
          >
            <a
              class="main-nav__link"
              href="/"
            >
              Каталог
            </a>
          </li>
          <li
            class="main-nav__item"
          >
            <a
              class="main-nav__link"
              href="#todo"
            >
              Гарантии
            </a>
          </li>
          <li
            class="main-nav__item"
          >
            <a
              class="main-nav__link"
              href="#todo"
            >
              Доставка
            </a>
          </li>
          <li
            class="main-nav__item"
          >
            <a
              class="main-nav__link"
              href="#todo"
            >
              О компании
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="form-search"
      >
        <form>
          <label>
            <svg
              aria-hidden="true"
              class="form-search__icon"
              height="16"
              width="16"
            >
              <use
                xlink:href="#icon-lens"
              />
            </svg>
            <input
              autocomplete="off"
              class="form-search__input"
              placeholder="Поиск по сайту"
              type="text"
            />
          </label>
          <ul
            class="form-search__select-list"
          >
            <li
              class="form-search__select-item"
              tabindex="0"
            >
              Cannonball Pro MX 8i
            </li>
            <li
              class="form-search__select-item"
              tabindex="0"
            >
              Cannonball Pro MX 7i
            </li>
            <li
              class="form-search__select-item"
              tabindex="0"
            >
              Cannonball Pro MX 6i
            </li>
            <li
              class="form-search__select-item"
              tabindex="0"
            >
              Cannonball Pro MX 5i
            </li>
            <li
              class="form-search__select-item"
              tabindex="0"
            >
              Cannonball Pro MX 4i
            </li>
          </ul>
        </form>
        <button
          class="form-search__reset"
          type="reset"
        >
          <svg
            aria-hidden="true"
            height="10"
            width="10"
          >
            <use
              xlink:href="#icon-close"
            />
          </svg>
          <span
            class="visually-hidden"
          >
            Сбросить поиск
          </span>
        </button>
      </div>
      <a
        class="header__basket-link"
        href="#todo"
      >
        <svg
          aria-hidden="true"
          height="16"
          width="16"
        >
          <use
            xlink:href="#icon-basket"
          />
        </svg>
      </a>
    </div>
  </header>
  <main>
    <div
      class="banner"
    />
    <div
      class="page-content"
    >
      <div
        class="breadcrumbs"
      >
        <div
          class="container"
        >
          <ul
            class="breadcrumbs__list"
          >
            <li
              class="breadcrumbs__item"
            >
              <a
                class="breadcrumbs__link"
                href="#todo"
              >
                Главная
                <svg
                  aria-hidden="true"
                  height="8"
                  width="5"
                >
                  <use
                    xlink:href="#icon-arrow-mini"
                  />
                </svg>
              </a>
            </li>
            <li
              class="breadcrumbs__item"
            >
              <span
                class="breadcrumbs__link breadcrumbs__link--active"
              >
                Каталог
              </span>
            </li>
          </ul>
        </div>
      </div>
      <section
        class="catalog"
      >
        <div
          class="container"
        >
          <h1
            class="title title--h2"
          >
            Каталог фото- и видеотехники
          </h1>
          <div
            class="page-content__columns"
          >
            <div
              class="catalog__aside"
            >
              <div
                class="catalog-filter"
              >
                <form
                  action="#"
                >
                  <h2
                    class="visually-hidden"
                  >
                    Фильтр
                  </h2>
                  <fieldset
                    class="catalog-filter__block"
                  >
                    <legend
                      class="title title--h5"
                    >
                      Цена, ₽
                    </legend>
                    <div
                      class="catalog-filter__price-range"
                    >
                      <div
                        class="custom-input"
                      >
                        <label>
                          <input
                            name="price"
                            placeholder="от"
                            type="number"
                          />
                        </label>
                      </div>
                      <div
                        class="custom-input"
                      >
                        <label>
                          <input
                            name="priceUp"
                            placeholder="до"
                            type="number"
                          />
                        </label>
                      </div>
                    </div>
                  </fieldset>
                  <fieldset
                    class="catalog-filter__block"
                  >
                    <legend
                      class="title title--h5"
                    >
                      Категория
                    </legend>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="photocamera"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Фотокамера
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="videocamera"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Видеокамера
                        </span>
                      </label>
                    </div>
                  </fieldset>
                  <fieldset
                    class="catalog-filter__block"
                  >
                    <legend
                      class="title title--h5"
                    >
                      Тип камеры
                    </legend>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="digital"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Цифровая
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="film"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Плёночная
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="snapshot"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Моментальная
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="collection"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Коллекционная
                        </span>
                      </label>
                    </div>
                  </fieldset>
                  <fieldset
                    class="catalog-filter__block"
                  >
                    <legend
                      class="title title--h5"
                    >
                      Уровень
                    </legend>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="zero"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Нулевой
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="non-professional"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Любительский
                        </span>
                      </label>
                    </div>
                    <div
                      class="custom-checkbox catalog-filter__item"
                    >
                      <label>
                        <input
                          name="professional"
                          type="checkbox"
                        />
                        <span
                          class="custom-checkbox__icon"
                        />
                        <span
                          class="custom-checkbox__label"
                        >
                          Профессиональный
                        </span>
                      </label>
                    </div>
                  </fieldset>
                  <button
                    class="btn catalog-filter__reset-btn"
                    type="reset"
                  >
                    Сбросить фильтры
                  </button>
                </form>
              </div>
            </div>
            <div
              class="catalog__content"
            >
              <div
                class="catalog-sort"
              >
                <form
                  action="#"
                >
                  <div
                    class="catalog-sort__inner"
                  >
                    <p
                      class="title title--h5"
                    >
                      Сортировать:
                    </p>
                    <div
                      class="catalog-sort__type"
                    >
                      <div
                        class="catalog-sort__btn-text"
                      >
                        <input
                          checked=""
                          id="sortPrice"
                          name="sort"
                          type="radio"
                        />
                        <label
                          for="sortPrice"
                        >
                          по цене
                        </label>
                      </div>
                      <div
                        class="catalog-sort__btn-text"
                      >
                        <input
                          id="sortPopular"
                          name="sort"
                          type="radio"
                        />
                        <label
                          for="sortPopular"
                        >
                          по популярности
                        </label>
                      </div>
                    </div>
                    <div
                      class="catalog-sort__order"
                    >
                      <div
                        class="catalog-sort__btn catalog-sort__btn--up"
                      >
                        <input
                          aria-label="По возрастанию"
                          id="up"
                          name="sort-icon"
                          type="radio"
                        />
                        <label
                          for="up"
                        >
                          <svg
                            aria-hidden="true"
                            height="14"
                            width="16"
                          >
                            <use
                              xlink:href="#icon-sort"
                            />
                          </svg>
                        </label>
                      </div>
                      <div
                        class="catalog-sort__btn catalog-sort__btn--down"
                      >
                        <input
                          aria-label="По убыванию"
                          checked=""
                          id="down"
                          name="sort-icon"
                          type="radio"
                        />
                        <label
                          for="down"
                        >
                          <svg
                            aria-hidden="true"
                            height="14"
                            width="16"
                          >
                            <use
                              xlink:href="#icon-sort"
                            />
                          </svg>
                        </label>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
              <div
                class="cards catalog__cards"
              >
                <div
                  class="product-card is-active"
                >
                  <div
                    class="product-card__img"
                  >
                    <picture>
                      <source
                        src="img/content/das-auge.webp"
                        srcset="img/content/das-auge@2x.webp"
                        type="image/webp"
                      />
                      <img
                        alt="Ретрокамера Dus Auge lV"
                        height="240"
                        src="img/content/das-auge.jpg"
                        srcset="img/content/das-auge@2x.jpg"
                        width="280"
                      />
                    </picture>
                  </div>
                  <div
                    class="product-card__info"
                  >
                    <div
                      class="rate product-card__rate"
                    >
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <p
                        class="visually-hidden"
                      >
                        Рейтинг: 3
                      </p>
                      <p
                        class="rate__count"
                      >
                        <span
                          class="visually-hidden"
                        >
                          Всего оценок:
                        </span>
                        16
                      </p>
                    </div>
                    <p
                      class="product-card__title"
                    >
                      Ретрокамера Dus Auge lV
                    </p>
                    <p
                      class="product-card__price"
                    >
                      <span
                        class="visually-hidden"
                      >
                        Цена:
                      </span>
                      200000
                    </p>
                  </div>
                  <div
                    class="product-card__buttons"
                  >
                    <button
                      class="btn btn--purple product-card__btn"
                      type="button"
                    >
                      Купить
                    </button>
                    <a
                      class="btn btn--transparent"
                      href="/product/1"
                    >
                      Подробнее
                    </a>
                  </div>
                </div>
                <div
                  class="product-card is-active"
                >
                  <div
                    class="product-card__img"
                  >
                    <picture>
                      <source
                        src="img/content/das-auge.webp"
                        srcset="img/content/das-auge@2x.webp"
                        type="image/webp"
                      />
                      <img
                        alt="Ретрокамера Dus Auge lV"
                        height="240"
                        src="img/content/das-auge.jpg"
                        srcset="img/content/das-auge@2x.jpg"
                        width="280"
                      />
                    </picture>
                  </div>
                  <div
                    class="product-card__info"
                  >
                    <div
                      class="rate product-card__rate"
                    >
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <p
                        class="visually-hidden"
                      >
                        Рейтинг: 3
                      </p>
                      <p
                        class="rate__count"
                      >
                        <span
                          class="visually-hidden"
                        >
                          Всего оценок:
                        </span>
                        16
                      </p>
                    </div>
                    <p
                      class="product-card__title"
                    >
                      Ретрокамера Dus Auge lV
                    </p>
                    <p
                      class="product-card__price"
                    >
                      <span
                        class="visually-hidden"
                      >
                        Цена:
                      </span>
                      150000
                    </p>
                  </div>
                  <div
                    class="product-card__buttons"
                  >
                    <button
                      class="btn btn--purple product-card__btn"
                      type="button"
                    >
                      Купить
                    </button>
                    <a
                      class="btn btn--transparent"
                      href="/product/1"
                    >
                      Подробнее
                    </a>
                  </div>
                </div>
                <div
                  class="product-card is-active"
                >
                  <div
                    class="product-card__img"
                  >
                    <picture>
                      <source
                        src="img/content/das-auge.webp"
                        srcset="img/content/das-auge@2x.webp"
                        type="image/webp"
                      />
                      <img
                        alt="Ретрокамера Dus Auge lV"
                        height="240"
                        src="img/content/das-auge.jpg"
                        srcset="img/content/das-auge@2x.jpg"
                        width="280"
                      />
                    </picture>
                  </div>
                  <div
                    class="product-card__info"
                  >
                    <div
                      class="rate product-card__rate"
                    >
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <p
                        class="visually-hidden"
                      >
                        Рейтинг: 3
                      </p>
                      <p
                        class="rate__count"
                      >
                        <span
                          class="visually-hidden"
                        >
                          Всего оценок:
                        </span>
                        16
                      </p>
                    </div>
                    <p
                      class="product-card__title"
                    >
                      Ретрокамера Dus Auge lV
                    </p>
                    <p
                      class="product-card__price"
                    >
                      <span
                        class="visually-hidden"
                      >
                        Цена:
                      </span>
                      50000
                    </p>
                  </div>
                  <div
                    class="product-card__buttons"
                  >
                    <button
                      class="btn btn--purple product-card__btn"
                      type="button"
                    >
                      Купить
                    </button>
                    <a
                      class="btn btn--transparent"
                      href="/product/1"
                    >
                      Подробнее
                    </a>
                  </div>
                </div>
                ;
              </div>
              <div
                class="pagination"
              >
                <ul
                  class="pagination__list"
                >
                  <li
                    class="pagination__item"
                  >
                    <a
                      class="pagination__link pagination__link--text"
                      href="#todo"
                    >
                      Назад
                    </a>
                  </li>
                  <li
                    class="pagination__item"
                  >
                    <a
                      class="pagination__link pagination__link--active"
                      href="#todo"
                    >
                      1
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </main>
  <footer
    class="footer"
  >
    <div
      class="container"
    >
      <div
        class="footer__info"
      >
        <a
          aria-label="Переход на главную"
          class="footer__logo"
          href="/"
        >
          <svg
            aria-hidden="true"
            height="36"
            width="100"
          >
            <use
              xlink:href="#icon-logo-mono"
            />
          </svg>
        </a>
        <p
          class="footer__description"
        >
          Интернет-магазин фото- и видеотехники
        </p>
        <ul
          class="social"
        >
          <li
            class="social__item"
          >
            <a
              aria-label="Переход на страницу вконтатке"
              class="link"
              href="#todo"
            >
              <svg
                aria-hidden="true"
                height="20"
                width="20"
              >
                <use
                  xlink:href="#icon-vk"
                />
              </svg>
            </a>
          </li>
          <li
            class="social__item"
          >
            <a
              aria-label="Переход на страницу pinterest"
              class="link"
              href="#todo"
            >
              <svg
                aria-hidden="true"
                height="20"
                width="20"
              >
                <use
                  xlink:href="#icon-pinterest"
                />
              </svg>
            </a>
          </li>
          <li
            class="social__item"
          >
            <a
              aria-label="Переход на страницу reddit"
              class="link"
              href="#todo"
            >
              <svg
                aria-hidden="true"
                height="20"
                width="20"
              >
                <use
                  xlink:href="#icon-reddit"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <ul
        class="footer__nav"
      >
        <li
          class="footer__nav-item"
        >
          <p
            class="footer__title"
          >
            Навигация
          </p>
          <ul
            class="footer__list"
          >
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="#todo"
              >
                Каталог
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="#todo"
              >
                Гарантии
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="#todo"
              >
                Доставка
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="#todo"
              >
                О компани
              </a>
            </li>
          </ul>
        </li>
        <li
          class="footer__nav-item"
        >
          <p
            class="footer__title"
          >
            Ресурсы
          </p>
          <ul
            class="footer__list"
          >
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="#todo"
              >
                Курсы операторов
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="#todo"
              >
                Блог
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="#todo"
              >
                Сообщество
              </a>
            </li>
          </ul>
        </li>
        <li
          class="footer__nav-item"
        >
          <p
            class="footer__title"
          >
            Поддержка
          </p>
          <ul
            class="footer__list"
          >
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="#todo"
              >
                FAQ
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="#todo"
              >
                Задать вопрос
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </footer>
</div>
`;
